homeassistant/util/unit_conversion.py

```diff
@@ -67,8 +70,41 @@ class BaseUnitConverter:
     @classmethod
     def convert(cls, value: float, from_unit: str | None, to_unit: str | None) -> float:
         """Convert one unit of measurement to another."""
+        return cls.converter_factory(from_unit, to_unit)(value)
```

Comment:
I make this use an LRU so that we don't take a performance hit in `sensor` when the units differ